src/_stories/wrap.py

```diff
@@ -21,37 +22,42 @@ def wrap_story(arguments, collected, cls_name, story_name, obj, spec, failures):
 
     methods = [(BeginningOfStory(cls_name, story_name), contract, protocol)]
 
-    for name in collected:
-
-        attr = getattr(obj, name)
+    for step in collected:
+
+        if isinstance(step, Parallel):
+            attr = step
+            attr.methods = [
```

Comment:
I don't understand why you decided to put instance bound methods and mounted stories into parallel marker which should be part of the collected spec. It should never leak into methods.

The same story could have different implementation details injected. And such an approach will not work at all.

For example,
```python
@dataclass
class Action:
    @story 
    def do(I):
        I.one
        I.two

    def one(self, ctx):
        x = self.do_one()
        return Success(x=x)

    def two(self, ctx):
        y = self.do_two()
        return Success(y=y)

    do_one: Callable
    do_two: Callable

Action(do_one=lambda: 1, do_two=lambda: 2).do()
Action(do_one=lambda: 3, do_two=lambda: 4).do()
```
At this point of complexity I think it will be reasonable to take a step back and think about methods structure.

We could write the `Methods` class in a separate module which will be able to iterate on story steps, handle history changes, and simplify working with the nested nature of the stories.

The wrap function will compose it the same way we do with a simple list right now. But handling code in the `execute_sequence` function I mentioned earlier.

In any case, execute modules should not know that such a thing as `MountedStory` even exists.

Reply:
That basically means we can get rid of markers module at all.

`Methods` class will not be a graph at this point, but a nested structure. And it could also say what strategy to use to execute the things it iterates on (`sequence`, `parallel`, `step`).

```python
def execute_sequence(runner, methods):
    result = None
    while True:
        executor_type, method = methods.next(result) # history modified by this method, skip marker is also processed here.
        result = executors[executor_type](runner, method)
```

Reply:
I don't care either way.
I'm satisfied with a working feature that I need to have right now.
Refactoring can be done post merge or even post release.

Reply:
Everything has a maintenance cost. Right now we have very limited resources to maintain the project. The project is already used by a few enterprise companies. In this condition, we have a limited choice.

> Refactoring can be done post-merge or even post-release.

Who would commit to doing this? At what time? If this promised refactoring will not happen in the promised time, can we unmerge the dirty implementation? I have a family to care about. I have a decent load on my daily job. I'll not make such commitment myself.

We don't have time for bad code. Each released feature should be implemented in a proper way. If a complex feature will be implemented in a half-done way, it'll not be released.

The only approach we can afford - reduce the scope of the feature without compromises in its implementation.